src/app/components/file-permissions-modal/file-permissions-modal.component.ts

```diff
@@ -0,0 +1,288 @@
+
+    modeStringChar = modeStringSym.charAt(0);
+    modeStringSym = modeStringSym.substring(1);
+    this.publicExecute = this.calcBooleanFromMode(modeStringChar);
+  }
+
+  calcBooleanFromMode(char: string): boolean {
+    if (char == '-') {
+      return false;
+    } else {
+      return true;
+    }
+  }
+
+  updateUI() {
+
+    let modeStringSym = "";
+
+    if (this.userExecute) { //1, 3, 5, or 7
+      if (this.userWrite) { //3 or 7
+        if (this.userRead) {
+          modeStringSym += "7";
+        } else {
+          modeStringSym += "3";
+        }
+      } else { // 1 or 5
+        if (this.userRead) {
+          modeStringSym += "5";
+        } else {
+          modeStringSym += "1";
+        }
+      }
+    } else { // 0, 2, 4, or 6
+      if (this.userWrite) { // 2 or 6
+        if (this.userRead) {
+          modeStringSym += "6";
+        } else {
+          modeStringSym += "2";
+        }
+      } else { //0 or 4
+        if (this.userRead) {
+          modeStringSym += "4";
+        } else {
+          modeStringSym += "0";
+        }
+      }
+    }
+
+    if (this.groupExecute) { //1, 3, 5, or 7
+      if (this.groupWrite) { //3 or 7
+        if (this.groupRead) {
+          modeStringSym += "7";
+        } else {
+          modeStringSym += "3";
+        }
+      } else { // 1 or 5
+        if (this.groupRead) {
+          modeStringSym += "5";
+        } else {
+          modeStringSym += "1";
+        }
+      }
+    } else { // 0, 2, 4, or 6
+      if (this.groupWrite) { // 2 or 6
+        if (this.groupRead) {
+          modeStringSym += "6";
+        } else {
+          modeStringSym += "2";
+        }
+      } else { //0 or 4
+        if (this.groupRead) {
+          modeStringSym += "4";
+        } else {
+          modeStringSym += "0";
+        }
+      }
+    }
+
+    if (this.publicExecute) { //1, 3, 5, or 7
+      if (this.publicWrite) { //3 or 7
+        if (this.publicRead) {
+          modeStringSym += "7";
+        } else {
+          modeStringSym += "3";
+        }
+      } else { // 1 or 5
+        if (this.publicRead) {
+          modeStringSym += "5";
+        } else {
+          modeStringSym += "1";
+        }
+      }
+    } else { // 0, 2, 4, or 6
+      if (this.publicWrite) { // 2 or 6
+        if (this.publicRead) {
+          modeStringSym += "6";
+        } else {
+          modeStringSym += "2";
+        }
+      } else { //0 or 4
+        if (this.publicRead) {
+          modeStringSym += "4";
+        } else {
+          modeStringSym += "0";
+        }
+      }
+    }
+
+    this.mode = parseInt(modeStringSym);
+    this.formatPermissions();
+  }
+
+  savePermissions() {
+    let url :string = ZoweZLUX.uriBroker.unixFileUri('chmod', this.path, undefined, undefined, undefined, false, undefined, undefined, undefined, this.mode.toString(), this.recursive);
+    this.http.post(url, null)
+    .map(res=>{
+      if (res.status == 200) {
```

Comment:
If I re-open this dialog after changing permissions the dialog still shows old permissions. Please update the permission info in the UI after you saved them in the filesystem.

Reply:
Yes, need to add this